monai/transforms/transform.py

```diff
@@ -61,18 +106,32 @@ def _apply_transform(
     Returns:
         ReturnType: The return type of `transform`.
     """
-    if isinstance(parameters, tuple) and unpack_parameters:
-        return transform(*parameters)
 
-    return transform(parameters)
+    lazy_tx = isinstance(transform, LazyTrait)
+
+    if lazy_tx is False or lazy is False:
+        _log_pending_info(data, transform, "Apply pending transforms", lazy, logger_name)
+        data = apply_pending_transforms(data, overrides)
+    elif lazy is None and transform.lazy is False:  # type: ignore[attr-defined]
+        _log_pending_info(data, transform, "Apply pending transforms", lazy, logger_name)
+        data = apply_pending_transforms(data, overrides)
+    else:
+        _log_pending_info(data, transform, "Accumulate pending transforms", lazy, logger_name)
```

Comment:
so now I think the only remaining debate (assuming it doesn't introduce any other debates) is here about 'correctness' and 'performance'. we can discuss it tomorrow at the dev meeting. cc @ericspod @atbenmurray @Nic-Ma